src/Mongo/src/Eventuous.Projections.MongoDB/Operations/InsertBuilder.cs

```diff
@@ -58,15 +52,16 @@ public class InsertManyBuilder : IMongoProjectorBuilder {
 
         Func<MessageConsumeContext<TEvent>, IEnumerable<T>> GetDocuments => Ensure.NotNull(_getDocuments, "Get documents function");
 
-        public InsertManyBuilder Documents(Func<TEvent, IEnumerable<T>> getDocuments) {
-            _getDocuments = ctx => getDocuments(ctx.Message)
-                .Select(x => x with { Position = ctx.GlobalPosition, StreamPosition = ctx.StreamPosition });
-
-            return this;
-        }
+        public InsertManyBuilder Documents(Func<TEvent, IEnumerable<T>> getDocuments) 
+            => Documents(ctx => getDocuments(ctx.Message));
 
         public InsertManyBuilder Documents(Func<MessageConsumeContext<TEvent>, IEnumerable<T>> getDocuments) {
-            _getDocuments = getDocuments;
+            _getDocuments = ctx => getDocuments(ctx)
+                .Select(x => x with 
+                {
+                    Position = ctx.GlobalPosition,
```

Comment:
I am not quite sure what this does, could you elaborate?

Reply:
I think it was a bug. One Document method set positions, while other didn't

Reply:
Ah, got this. You're completely right.